packages/calling-stateful-client/src/TeamsCallAgentDeclarative.ts

```diff
@@ -66,43 +67,45 @@ class ProxyTeamsCallAgent extends ProxyCallAgentCommon implements ProxyHandler<D
     throw new Error('Not reachable code, DeclarativeTeamsCallAgent.callDeclaratify must be called with an TeamsCall.');
   }
 
-  protected startCall(agent: CallAgentCommon, args: Parameters<TeamsCallAgent['startCall']>): CallCommon {
+  protected startCall(agent: TeamsCallAgent, args: Parameters<TeamsCallAgent['startCall']>): CallCommon {
     /* @conditional-compile-remove(teams-identity-support) */
     if (_isTeamsCallAgent(agent)) {
       return agent.startCall(...args);
     }
     throw new Error('Not reachable code, DeclarativeTeamsCallAgent.startCall must be called with an TeamsCallAgent.');
   }
 
-  protected joinCall(agent: CallAgentCommon, args: Parameters<TeamsCallAgent['join']>): CallCommon {
+  protected joinCall(agent: TeamsCallAgent, args: Parameters<TeamsCallAgent['join']>): CallCommon {
     /* @conditional-compile-remove(teams-identity-support) */
     if (_isTeamsCallAgent(agent)) {
```

Comment:
After this if sentence the agent instance will be TeamsCallAgent, which means we don't need force cast at level of function parameter